### PR TITLE
Add port of Meta's Sapiens2 dense prediction model(s)

### PR DESCRIPTION
**Pull request by eklipse2k8**
MLX port of [Sapiens2](https://github.com/facebookresearch/sapiens2) (Meta, ICLR 2026) — high-resolution vision transformers pretrained on 1B human images for human-centric dense prediction. Unlike standard VLMs, these models output dense predictions instead of text.

Supported tasks (inference only):

| Task | Output | HF repos |
|------|--------|----------|
`facebook/sapiens2-pose-{0.4b,0.8b,1b,5b}` | | Segmentation (29 parts) | per-pixel class logits | `facebook/sapiens2-seg-{0.4b,0.8b,1b,5b}` | | Surface normals | per-pixel normals | `facebook/sapiens2-normal-{0.4b,0.8b,1b,5b}` | | Pointmaps | per-pixel XYZ + scale | `facebook/sapiens2-pointmap-{0.4b,0.8b,1b,5b}` | | Matting | alpha + foreground | `facebook/sapiens2-matting-1b` |

The official HF repos load directly — `config.json` (`model_type: "sapiens2"`) and the safetensors weights need no conversion.

### CODE REVIEW

File: mlx_vlm/models/sapiens2/__init__.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+"""Sapiens2 for MLX: dense-prediction vision transformers, inference only.
+
+See README.md for usage and model repos.
+"""
+
+from . import image  # Install processor patch
```
Comment (Lazarus-931) [inline mlx_vlm/models/sapiens2/__init__.py]: this should be something like:

```
__init__ = ["sapiens2", .....]
```

and so on

**Review by lucasnewman — APPROVED** [review]
@eklipse2k8 Awesome, thanks!
